boto/connection.py

```diff
@@ -1118,7 +1118,7 @@ def _get_correct_s3_endpoint_from_response(self, request,
         elif err.error_code == 'IllegalLocationConstraintException':
             region_regex = (
                 'The (.*?) location constraint is incompatible for the region '
```

Comment:
Why do we not have an `r` marked here ?

Reply:
As there is no backslash character () in the string. warnings are raised only for an invalid backslash character.

Reply:
Let's not be driven by warnings alone. If the standard has changed to prefix `r` AND if we are making changes in the same method, then let's make it here as well for consistencies.

Reply:
done.